### PR TITLE
Simplify example with single layer class

### PR DESCRIPTION
**Pull request by streeve**
Moves more code into source - can also be extended for multi-layer problems 

Also add convenience top-level header

### CODE REVIEW

File: src/Finch_Grid.hpp [hunk 1]
```diff
@@ -146,17 +146,17 @@ class Grid
 
     auto getLocalGrid() { return local_grid; }
 
-    auto getIndexSpace()
+    auto getIndexSpace() const
```
Comment (MattRolchigo) [inline src/Finch_Grid.hpp:149]: Was adding const to these functions a related change to get this to compile?

File: src/Finch_Run.hpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+#include "Finch_Inputs.hpp"
+#include "Finch_SolidificationData.hpp"
+#include "Finch_Solver.hpp"
+#include "MovingBeam/Finch_MovingBeam.hpp"
+
+namespace Finch
+{
+
+template <typename MemorySpace>
+class SingleLayer
```
Comment (MattRolchigo) [inline src/Finch_Run.hpp:28]: I don't know if there are any plans to make finch into a code that simulates multiple layers with heat accumulation from prior simulated layers (@colemanjs can correct me if I'm wrong here), but the name `SingleLayer` is a little confusing because it implies that we're simulating/can simulate multiple layers when we aren't. Even with future in-memory coupling using ExaCA, we'd only simulate one layer at a time, I feel like something like `Layer` might be a better name? This is probably just me being picky though

File: src/Finch_SolidificationData.hpp [hunk 3]
```diff
@@ -83,23 +96,15 @@ class SolidificationData
...
 
-        // get values from simulation data bases
-        auto liquidus = db.properties.liquidus;
-        auto dt = db.time.time_step;
-        auto time = db.time.time;
-        auto cell_size = db.space.cell_size;
-
         // get local copies of member variables on device for KOKKOS_LAMBDA
         auto local_count = count;
         auto local_capacity = capacity;
```
Comment (MattRolchigo) [inline src/Finch_SolidificationData.hpp:110]: Are these lines still necessary with the change to `KOKKOS_CLASS_LAMBDA` below?

File: src/Finch_SolidificationData.hpp [hunk 4]
```diff
@@ -83,23 +96,15 @@ class SolidificationData
         auto tm =
             Cabana::Grid::createArray<double, memory_space>( "tm", layout );
         tm_view = tm->view();
-
-        folder_name = db.sampling.directory_name;
     }
 
-    void updateEvents()
+    void updateEvents( Grid<memory_space>& grid )
```
Comment (MattRolchigo) [inline src/Finch_SolidificationData.hpp:101]: Why this change instead of storing `local_mesh` locally in the solidification class?

File: src/Finch_Grid.hpp (hunk 1 above)
Comment (streeve) [reply src/Finch_Grid.hpp:149]: I believe that was the case - let me check

File: src/Finch_Run.hpp (hunk 2 above)
Comment (streeve) [reply src/Finch_Run.hpp:28]: I agree - fixed

File: src/Finch_SolidificationData.hpp (hunk 4 above)
Comment (streeve) [reply src/Finch_SolidificationData.hpp:101]: I was trying to separate these classes and I figured since we have to extract so many other things may as well not store just some of it (like the the `local_mesh`

**Review by MattRolchigo — APPROVED** [review]
Looks good other than the CI thing

File: src/Finch_Grid.hpp (hunk 1 above)
Comment (streeve) [reply src/Finch_Grid.hpp]: Removed

**Comment by streeve** [comment]
Intel failure unrelated
